Receive loop still propagates IOException-as-cause to consumers after close

## Problem

Reported by konstructor during ksrpc 1.0.0-RC5 integration validation. The fix in PR #188 (issue #187) downgraded the *ksrpc-internal* receive-loop log when \`isClosed\`, but the underlying exception still propagates to consumers awaiting on the multiChannel, who then log it themselves at WARN.

konstructor's observation (paraphrased):

> #188 quieted the ksrpc-internal log statement, but the exception still propagates out of ksrpc to the caller, and the caller (us) logs the propagated value. The fix only quiets ksrpc's own loggers, not consumers who catch and log the throwable.

Concrete shape on konstructor's adolin deployment under RC5:
\`\`\`
WARN ScriptManager - java.util.concurrent.CancellationException: Multi-channel failure
   at com.monkopedia.ksrpc.packets.internal.PacketChannelBase.executeReceive(PacketChannelBase.kt:194)
   ...
   Caused by: java.io.IOException: ... RawSourceChannel ... is closed for reading
   at com.monkopedia.ksrpc.sockets.internal.PacketUtilsKt.readFields(PacketUtils.kt:43)
\`\`\`

Same volume as the pre-#188 RC4 noise (~169 WARN events per 2-test run), just different exception type. Not a functional bug — konstructor is fine on RC5 — but the log-noise improvement from #188 is essentially zero for consumers who log on caught exceptions.

## Root cause

\`PacketChannelBase.executeReceive\` catches the receive-loop exception, downgrades the internal log when \`isClosed\`, and then unconditionally calls:

\`\`\`kotlin
binaryChannels.values.forEach { it.closeWithError(t) }
multiChannel.close(CancellationException(\"Multi-channel failure\", t))
\`\`\`

That \`CancellationException\` carries the IOException as cause, and surfaces to anyone awaiting on the multiChannel — which is the consumer-side log.

## Fix

When \`isClosed\` (consumer initiated the teardown), close the multiChannel and binaryChannels cleanly (no cause). Consumers awaiting on them see a normal close instead of a CancellationException-wrapping-IOException. Pre-close exceptions still propagate as before, since those *are* real failures.

\`\`\`kotlin
if (isClosed) {
    env.logger.debug(\"MultiChannel\", \"Receive loop ended after close\", t)
    binaryChannels.values.forEach { it.closeWithError(null) }
    multiChannel.close()
} else {
    env.logger.warn(\"MultiChannel\", \"Exception in multichannel\", t)
    binaryChannels.values.forEach { it.closeWithError(t) }
    multiChannel.close(CancellationException(\"Multi-channel failure\", t))
}
\`\`\`

## Severity

Cosmetic / observability, but blocks 1.0.0 per user direction (\"if we have known issues we can resolve, I'd like to resolve them\").

--- a/ksrpc-packets/src/commonMain/kotlin/PacketChannelBase.kt
+++ b/ksrpc-packets/src/commonMain/kotlin/PacketChannelBase.kt
@@ -182,16 +182,22 @@ abstract class PacketChannelBase<T>(
             // Once close() has flipped isClosed, the peer is gone by definition — the read
             // path will see a closed channel and surface IOException ("closed for reading"
             // from PacketUtils.readFields, or framework-equivalent). That's normal teardown,
-            // not a failure: downgrade to debug so it doesn't read as a scary WARN in
-            // consumer logs. Mirrors the close-during-write handling in
-            // ksrpc-sockets/InputOutputStreams.copyToAndFlush (#169 / PR #172).
+            // not a failure: downgrade the local log to debug AND close the multiChannel /
+            // binaryChannels cleanly so consumers awaiting on them see a normal close
+            // instead of a CancellationException wrapping the IOException. Mirrors the
+            // close-during-write handling in ksrpc-sockets/InputOutputStreams.copyToAndFlush
+            // (#169 / PR #172). #187's PR #188 only quieted the ksrpc-internal log; the
+            // CancellationException was still propagating to consumers (konstructor's
+            // ScriptManager surfaced it at WARN with the IOException cause attached).
             if (isClosed) {
                 env.logger.debug("MultiChannel", "Receive loop ended after close", t)
+                binaryChannels.values.forEach { it.closeWithError(null) }
+                multiChannel.close()
             } else {
                 env.logger.warn("MultiChannel", "Exception in multichannel", t)
+                binaryChannels.values.forEach { it.closeWithError(t) }
+                multiChannel.close(CancellationException("Multi-channel failure", t))
             }
-            binaryChannels.values.forEach { it.closeWithError(t) }
-            multiChannel.close(CancellationException("Multi-channel failure", t))
         }
     }
 

